Add a new Settings page in the Client side section

gremlation opened the pull request:
The sample code provided on the Examples page uses a deprecated way of adding settings and is missing a lot of details.

The new page shows the new way of doing things and explains the different data types, options, and events.

christian-byrne commented:
[New Settings API](https://github.com/Comfy-Org/ComfyUI_frontend#:~:text=v1.3.22%3A%20New%20settings%20API)

```ts
// Register a new setting
app.registerExtension({
  name: 'TestExtension1',
  settings: [
    {
      id: 'TestSetting',
      name: 'Test Setting',
      type: 'text',
      defaultValue: 'Hello, world!'
    }
  ]
})

// Get the value of a setting
const value = app.extensionManager.setting.get('TestSetting')

// Set the value of a setting
app.extensionManager.setting.set('TestSetting', 'Hello, universe!')
```

christian-byrne commented:
[Setting Type](https://github.com/Comfy-Org/ComfyUI_frontend/blob/74b02237dbc8947667134776e3c731c6d8c9242b/src/types/settingTypes.ts#L4)

```ts
export interface SettingParams extends FormItem {
  id: keyof Settings
  defaultValue: any | (() => any)
  onChange?: (newValue: any, oldValue?: any) => void
  // By default category is id.split('.'). However, changing id to assign
  // new category has poor backward compatibility. Use this field to overwrite
  // default category from id.
  // Note: Like id, category value need to be unique.
  category?: string[]
  experimental?: boolean
  deprecated?: boolean
  // Deprecated values are mapped to new values.
  migrateDeprecatedValue?: (value: any) => any
  // Version of the setting when it was added
  versionAdded?: string
  // Version of the setting when it was last modified
  versionModified?: string
}

/**
 * The base form item for rendering in a form.
 */
export interface FormItem {
  name: string
  type: SettingInputType | SettingCustomRenderer
  tooltip?: string
  attrs?: Record<string, any>
  options?: Array<string | SettingOption>
}
```

christian-byrne commented:
There's also colorpicker and file upload components.

You can bind properties to the components with `attrs`. There is full documentation on all the components in [Primevue docs].

For example This is the default number:

```ts
app.registerExtension({
  name: 'TestExtension4',
  settings: [
    {
      id: 'TestSetting222222',
      category: ['Test2','d'],
      name: 'Test Settingd',
      type: 'number',
      defaultValue: 22,
    }
  ]
})
```

![Selection_680](https://github.com/user-attachments/assets/1c9d580a-016a-4278-8b15-031213ba48ed)

You can look at [`InputNumber` documentation](https://primevue.org/inputnumber/) to find props you want to use and then apply them:

```ts
app.registerExtension({
  name: 'TestExtension4',
  settings: [
    {
      id: 'TestSetting222222',
      category: ['Test2','d'],
      name: 'Test Settingd',
      type: 'number',
      defaultValue: 22,
      attrs: {showButtons: true, prefix: '$'}
    }
  ]
})
```

![Selection_679](https://github.com/user-attachments/assets/0f0f08ad-ccdb-4af4-b914-f673dffe1691)

Btw, your explanations and writing are very good on these docs.

gremlation commented:
Thanks for the information @christian-byrne! I added references to the relevant components and updated some of the information that was incorrect, so it explains how to support floating point numbers in number inputs for example.

I wasn't able to figure out how to add a color picker or file input - are these definitely supported for settings, or are these only supported as widgets inside nodes? Do you have a working example?

gremlation commented:
Ah, got it. Thanks, I'll follow up with another PR to add those docs when it's in stable then.

christian-byrne commented:
Sounds good. Were you able to try using `app.extensionManager.setting.get` instead of `app.ui.settings.getSettingValue`? The later is marked as deprecated.

gremlation commented:
Sorry, I didn't see that the first time around. It was marked as deprecated five days ago, my working copy didn't have those changes in. I've updated it now.

christian-byrne commented:
It looks good to me! I just did console test with all the examples and they all work. Just one missing comma [here](https://github.com/Comfy-Org/docs/blob/fbc04c8f676d00ffd41ed3636bb42d2f5060f81c/essentials/javascript_settings.mdx?plain=1#L217-L221).

You could also consider documenting how the `onChange` callback is invoked immediately after registration. Someone on Disc specifically asked for it to be documented a while ago. Up to you!

gremlation commented:
Thanks! Fixed the comma and added a note about the change event.

robinjhuang commented:
Thanks for adding this!
